subsys/net/openthread/Kconfig

```diff
@@ -19,12 +19,46 @@ config NORDIC_SECURITY_PROMPTLESS
 choice OPENTHREAD_SECURITY
 	depends on !OPENTHREAD_SECURITY_INTERNAL
 
+config OPENTHREAD_NRF_SECURITY_PSA_CHOICE
+	bool "nRF Security with PSA crypto enabled"
+	select OPENTHREAD_NRF_SECURITY_PSA
+
 config OPENTHREAD_NRF_SECURITY_CHOICE
 	bool "nRF Security"
 	select OPENTHREAD_NRF_SECURITY
 
 endchoice
 
+config OPENTHREAD_NRF_SECURITY_PSA
+	bool
+	select EXPERIMENTAL
+	select NRF_SECURITY
+	select MBEDTLS_ENABLE_HEAP
+	select MBEDTLS_TLS_LIBRARY
+	select MBEDTLS_PSA_CRYPTO_STORAGE_C if !BUILD_WITH_TFM
+	select PSA_NATIVE_ITS if !BUILD_WITH_TFM
+	select PSA_WANT_ALG_HKDF
+	select PSA_WANT_ALG_CCM
+	select PSA_WANT_ALG_CMAC
+	select PSA_WANT_ALG_HMAC
+	select PSA_WANT_ALG_ECB_NO_PADDING
+	select PSA_WANT_ALG_SHA_256
+	select PSA_WANT_ALG_JPAKE
+	select PSA_WANT_ALG_TLS12_ECJPAKE_TO_PMS
+	select PSA_WANT_ALG_DETERMINISTIC_ECDSA if OPENTHREAD_ECDSA
+	select PSA_WANT_ALG_ECDSA if OPENTHREAD_ECDSA
+	select PSA_WANT_ALG_ECDH if OPENTHREAD_ECDSA
+	select PSA_WANT_ECC_SECP_R1_256
+	select PSA_WANT_GENERATE_RANDOM
+	select PSA_WANT_KEY_TYPE_AES
+	select PSA_WANT_KEY_TYPE_ECC_PUBLIC_KEY
+	select PSA_WANT_KEY_TYPE_ECC_KEY_PAIR
```

Comment:
Is the ECC curve and KEY only needed with OPENTHREAD_ECDSA enabled maybe?

Reply:
If I remove direct selection of `PSA_WANT_KEY_TYPE_ECC_PUBLIC_KEY` and make `PSA_WANT_KEY_TYPE_ECC_KEY_PAIR` dependend on `OPENTHREAD_ECDSA`, I'd need to make the same dependency for `PSA_WANT_ECC_SECP_R1_256` . Then I'll get build errors:
```
   77 | #error "MBEDTLS_DHM_C defined, but not all prerequisites"
      |  ^~~~~
/home/mbaczmanski/ncs/nrf/ext/oberon/psa/core/include/mbedtls/check_config.h:148:2: error: #error "MBEDTLS_ECP_C defined, but not all prerequisites"
  148 | #error "MBEDTLS_ECP_C defined, but not all prerequisites"
      |  ^~~~~
/home/mbaczmanski/ncs/nrf/ext/oberon/psa/core/include/mbedtls/check_config.h:327:2: error: #error "MBEDTLS_KEY_EXCHANGE_ECJPAKE_ENABLED defined, but not all prerequisites"
  327 | #error "MBEDTLS_KEY_EXCHANGE_ECJPAKE_ENABLED defined, but not all prerequisites"
      |  ^~~~~
/home/mbaczmanski/ncs/nrf/ext/oberon/psa/core/include/mbedtls/check_config.h:989:2: error: #error "MBEDTLS_X509_USE_C defined, but not all prerequisites"
  989 | #error "MBEDTLS_X509_USE_C defined, but not all prerequisites"
```
which would force me to disable joiner and coaps. We have added a note to next step of PSA crypto API support to take a deeper look into all dependencies. OPENTHREAD_ECDSA is onef them as it isn't directly correlated with commissioner/joiner/COAPS configs, but joiner is eabled in all library configurations for now, so it needs to be always selected. I think we could keep this one the way it is for experimental support and optimize it later?

Reply:
Ok, if you get check_config errors then lets go with what works :) It can be a future improvement.